Ensembl/ensembl-client
# Add screens for invalid urls

## Description
This PR addresses several types of invalid urls:
- a genome with a given id may be missing
- the id of a feature in the url may not conform to the required format (e.g. does not have the feature type prefix, such as `gene:` before the stable id)
- the id of a feature, while formatted correctly, may not correspond to any existing feature 

The PR introduces error screens, as designed in [this XD](https://xd.adobe.com/view/b6d9a69d-468f-44b1-9333-419e228aa5ff-e38a/screen/90d1e50f-90fa-4586-83a2-99bd2232e4b5). For Entity Viewer and Species page, the server-side code will return a 404 status if a genome or a feature does not exist.

### Warning
Make sure relevant environment variables are set in production before this gets deployed. 

TODO:
- [x] Add screens for missing genome and missing focus object on the genome browser page
- [x] Remove saved species with invalid genome ids
- [ ] Show error status for species lozenges

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1617

## Deployment URL(s)
http://missing-genomes.review.ensembl.org

## How to test
1. Use an arbitrary string for a genome id on the Species page, Genome Browser page, and Entity Viewer page
2. Manually, save a species with an invalid `genome_id` and `url_slug` in the local storage:

```js
const x = [
    {
        "genome_id": "homo_sapiens_GCA_000001405_28",
        "common_name": "Human",
        "scientific_name": "Homo sapiens",
        "assembly_name": "GRCh38.p13",
        "url_slug": "grch38",
        "isEnabled": true
    },
    {
        "genome_id": "foobar",
        "common_name": "Hello",
        "scientific_name": "Lorem ipsum",
        "assembly_name": "foo-bar",
        "url_slug": "foo",
        "isEnabled": true
    }
];
window.localStorage.setItem('speciesSelector.selectedSpecies', JSON.stringify(x))
```
Then use the value of the genome_id or the url_slug field to create urls for this invalid species. Click on the "Continue" button, and make sure that the invalid species disappears from among the species lozenges

3. Generate urls with invalid focus object or entity ids for Genome Browser and Entity Viewer. Make sure you can see the page informing you that the feature is unavailable, and that you can navigate away from that screen. 

## Views affected
- Species page
- Entity Viewer
- Genome Browser
